src/Native/Runtime/unix/PalRedhawkUnix.cpp

```diff
@@ -1019,16 +941,17 @@ extern "C" UInt32_BOOL DuplicateHandle(
     return lpTargetHandle != nullptr;
 }
 
-// TODO: what to do with this? The CRITICAL_SECTION at the other side is different!
-typedef struct _RTL_CRITICAL_SECTION {
-    pthread_mutex_t mutex;
-} CRITICAL_SECTION, RTL_CRITICAL_SECTION, *PRTL_CRITICAL_SECTION;
+extern "C" UInt32_BOOL InitializeCriticalSection(CRITICAL_SECTION * lpCriticalSection)
+{
+    return pthread_mutex_init(&lpCriticalSection->mutex, NULL) == 0;
+}
```

Comment:
This change is fixing a memory corruption I am seeing on Mac, so it will be great to see it it committed soon :)